Makefile

```diff
@@ -258,14 +258,13 @@ uninstall:
 
 # find or download controller-gen
 # download controller-gen if necessary
-controller-gen:
-ifeq (, $(shell which controller-gen))
-	GO111MODULE=on go get sigs.k8s.io/controller-tools/cmd/controller-gen@v0.3.0
-	go mod tidy
-CONTROLLER_GEN=$(GOBIN)/controller-gen
-else
-CONTROLLER_GEN=$(shell which controller-gen)
-endif
+controller-gen: __tooling-image
+CONTROLLER_GEN=docker run -it -v $(shell pwd):/gatekeeper gatekeeper-tooling controller-gen
+
+__tooling-image:
```

Comment:
is the double-underscore intended to signify non-user-facing targets? If so, I like it.

Reply:
yes, exactly

Reply:
Very nice, I could see us using this to hide a bunch of our current Makefile targets